caas/kubernetes/provider/k8s_test.go

```diff
@@ -3277,6 +3289,186 @@ func (s *K8sBrokerSuite) TestEnsureServiceForDeploymentWithDevices(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 }
 
+func (s *K8sBrokerSuite) TestEnsureServiceForDaemonSetWithDevicesCreate(c *gc.C) {
```

Comment:
Can we include a test with constraint tags so that we test that node affinity is set etc